### PR TITLE
Added geom_text_repel and geom_label_repel from ggrepel package into the R-Instat list of geoms 

### LINKED ISSUE #6626: Add geoms from ggrepel into the R-Instat list of geoms
@lilyclements that's a good observation in #6523 .  @shadrackkibet is going to be adding labels in the bar chart dialogue, once the dialogue has included the other options.  He is also adding a new feature into the label and text geoms.  

We have been using ggrepel in our climatic mapping dialogues, so the package is installed.  We should ass gg_label_repel and gg_text_repel into the list of geoms.  @HawardKetoyoMsatsi now seems to be able to do this very quickly, so I am suggesting this for him too.  Haward, I hope that's ok? 

I am not sure whether they are usually needed for the bar chart labels, but it could be good to investigate once they are available.

### PR DESCRIPTION
**Pull request by HawardKetoyoMsatsi**
This Fixes #6626
@rdstern  this is ready for review.
@africanmathsinitiative/developers this is ready for review

### CODE REVIEW

**Review by rdstern — CHANGES_REQUESTED** [review]
@HawardKetoyoMsatsi that looks promising.  But is gives an error.  It generates the command ggplot2::geom_label_repel when it should be ggrepel instead.  Same for text _repel.

**Comment by HawardKetoyoMsatsi** [comment]
@rdstern the above changes have been made.

**Comment by HawardKetoyoMsatsi** [comment]
@rdstern could you please review this, It will be needed to be merged for the ggrepel to be used to solve another issue.

**Review by rdstern — DISMISSED** [review]

File: instat/ucrGeom.vb [hunk 1]
```diff
@@ -1242,6 +1244,51 @@ Public Class ucrGeom
...
+        clsgeom_label_repel.AddLayerParameter("nudge_y", "numeric", "0", lstParameterStrings:={2})
+        clsgeom_label_repel.AddLayerParameter("show.legend", "list", "TRUE", lstParameterStrings:={"NA", "TRUE", "FALSE"})
+        clsgeom_label_repel.AddLayerParameter("xlim", "numeric", "1", lstParameterStrings:={2})
+        clsgeom_label_repel.AddLayerParameter("ylim", "numeric", "1", lstParameterStrings:={2})
+        clsgeom_label_repel.AddLayerParameter("na.rm", "list", "FALSE", lstParameterStrings:={"TRUE", "FALSE"})
+        clsgeom_label_repel.AddLayerParameter("seed", "editablelist", "NA", lstParameterStrings:={"NA"})
+        clsgeom_label_repel.AddLayerParameter("verbose", "list", "False", lstParameterStrings:={"True", "False"})
+        clsgeom_label_repel.AddLayerParameter("inherit.aes", "list", "True", lstParameterStrings:={"True", "False"})
+        'clsgeom_label_repel.AddLayerParameter("direction", "list", Chr(34) & "c"(Chr(34) & "both" & Chr(34), Chr(34) & "x" & Chr(34), Chr(34) & "y" & Chr(34)) & Chr(34), lstParameterStrings:={Chr(34) & "both" & Chr(34), Chr(34) & "x" & Chr(34), Chr(34) & "y" & Chr(34)})
```
Comment (shadrackkibet) [inline instat/ucrGeom.vb:1289]: Why is this commented out?

File: instat/ucrGeom.vb [hunk 2]
```diff
@@ -1831,6 +1878,49 @@ Public Class ucrGeom
...
+        clsgeom_text_repel.AddLayerParameter("max.time", "numeric", "0.5")
+        clsgeom_text_repel.AddLayerParameter("max.iter", "numeric", "1000", lstParameterStrings:={0, 1})
+        clsgeom_text_repel.AddLayerParameter("max.overlaps", "numeric", "10", lstParameterStrings:={0, 1})
+        clsgeom_text_repel.AddLayerParameter("nudge_x", "numeric", "0", lstParameterStrings:={2})
+        clsgeom_text_repel.AddLayerParameter("nudge_y", "numeric", "0", lstParameterStrings:={2})
+        clsgeom_text_repel.AddLayerParameter("show.legend", "list", "TRUE", lstParameterStrings:={"NA", "TRUE", "FALSE"})
+        clsgeom_text_repel.AddLayerParameter("xlim", "numeric", "1", lstParameterStrings:={2})
+        clsgeom_text_repel.AddLayerParameter("ylim", "numeric", "1", lstParameterStrings:={2})
+        clsgeom_text_repel.AddLayerParameter("na.rm", "list", "FALSE", lstParameterStrings:={"TRUE", "FALSE"})
+        'clsgeom_text_repel.AddLayerParameter("direction", "list", Chr(34) & "c"(Chr(34) & "both" & Chr(34), Chr(34) & "x" & Chr(34), Chr(34) & "y" & Chr(34)) & Chr(34), lstParameterStrings:={Chr(34) & "both" & Chr(34), Chr(34) & "x" & Chr(34), Chr(34) & "y" & Chr(34)})
```
Comment (shadrackkibet) [inline instat/ucrGeom.vb:1918]: Also here

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

**Comment by HawardKetoyoMsatsi** [comment]
@shadrackkibet the above is fixed, hovever due to the size of the ialog cant allow all parameters to be vissible
(inherit.aes,verbose,seeddirection) wont show for geom_text_repel while (show.legend, na.rm, seed, inherit.aes, verbose, direction) wont show for geom_label_repel

**Comment by shadrackkibet** [comment]
Just before we merge this. Could you check with @maxwellfundi on the criteria he adopted before in choosing the number parameters to add? I guess the choice is based on those parameters that are used "mostly" which will need some research online. There is no need to add parameters that won't be visible in the dialog!

**Review by rdstern — APPROVED** [review]

**Comment by HawardKetoyoMsatsi** [comment]
@shadrackkibet I updated the order and removed the extra parameters that cant apper whe R-instat is run

**Review by shadrackkibet — APPROVED** [review]
@rdstern could you check and re-approve?

**Review by rdstern — APPROVED** [review]

**Pull request merged**